From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: chickeneer <emcchickeneer@gmail.com>
Date: Sun, 17 May 2020 15:40:10 -0500
Subject: [PATCH] Do not require nether fortress structure for mob spawning

Due to failed fortress conversions - some structure data is missing.
This allows players to make their own fortress with nether bricks

diff --git a/src/main/java/net/minecraft/world/level/NaturalSpawner.java b/src/main/java/net/minecraft/world/level/NaturalSpawner.java
index 55b937f802ee7066cb13b9a497932038b2905ff0..260ca6ccb76f754be6bcf0842b6485886cf6f360 100644
--- a/src/main/java/net/minecraft/world/level/NaturalSpawner.java
+++ b/src/main/java/net/minecraft/world/level/NaturalSpawner.java
@@ -404,7 +404,7 @@ public final class NaturalSpawner {
     }
 
     private static WeightedRandomList<MobSpawnSettings.SpawnerData> mobsAt(ServerLevel world, StructureFeatureManager structureAccessor, ChunkGenerator chunkGenerator, MobCategory spawnGroup, BlockPos pos, @Nullable Biome biome) {
-        return spawnGroup == MobCategory.MONSTER && world.getBlockState(pos.below()).is(Blocks.NETHER_BRICKS) && structureAccessor.getStructureAt(pos, false, StructureFeature.NETHER_BRIDGE).isValid() ? StructureFeature.NETHER_BRIDGE.getSpecialEnemies() : chunkGenerator.getMobsAt(biome != null ? biome : world.getBiome(pos), structureAccessor, spawnGroup, pos);
+        return spawnGroup == MobCategory.MONSTER && world.getWorld().getEnvironment() == org.bukkit.World.Environment.NETHER && world.getBlockState(pos.below()).is(Blocks.NETHER_BRICKS) /*&& structureAccessor.getStructureAt(pos, false, StructureFeature.NETHER_BRIDGE).isValid()*/? StructureFeature.NETHER_BRIDGE.getSpecialEnemies() : chunkGenerator.getMobsAt(biome != null ? biome : world.getBiome(pos), structureAccessor, spawnGroup, pos); // EMC
     }
 
     private static BlockPos getRandomPosWithin(Level world, LevelChunk chunk) {
